codegen/java-gen/src/main/scala/kalix/codegen/java/ComponentsSourceGenerator.scala

```diff
@@ -141,7 +141,7 @@ object ComponentsSourceGenerator {
              |public DeferredCall<$inputType, $outputType> $commandMethod($inputType $paramName) {
              |  return new GrpcDeferredCall<>(
              |    ${lowerFirst(command.inputType.name)},
-             |    MetadataImpl.Empty(),
+             |    (MetadataImpl) ((InternalContext) context).componentGrpcClientMetadata(),
```

Comment:
Why are the casts needed, isn't `context` defined as an `InternalContext`? Also, would be fine to mark `componentGrpclClientMetadata` as `MetadataImpl` to start with, since anyway internal.